### PR TITLE
Disable config update to SmallryeMetricsCDIMetadata and issue Info message.

### PR DESCRIPTION
**Pull request by Channyboy**
Update service-component to use @Modified annotation over the atomic boolean and issue info about libraryRef changes only taking effect after server restart

For #23314 

#build

### CODE REVIEW

**Review by fmhwong — DISMISSED** [review]
LGTM

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 1]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
 internal.error.CWMMC0006E.explanation=The application server experienced an internal error.
 internal.error.CWMMC0006E.useraction=Gather a set of traces and open a new issue against your application server.
 
+#Information regarding libraryRef attribute taking effect only after server restart
+configurationChange.info.CWMMC0007I=CWMMC0007I: A configuration change was detected for the MicroProfile Metrics feature. If a changes was made to the libraryRef attribute then a server restart is required for the changes to take effect.
+configurationChange.info.CWMMC0007I.explanation=A configuration change was detected for the MicroProfile Metrics feature. If a changes was made to the libraryRef attribute then a server restart is required for the changes to take effect.
```
Comment (donbourne) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: a changes -> a change
the changes -> the change

Comment (donbourne) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: explanation should say something about how the mpMetrics feature does not allow for dynamic replacement of the library as applications could have registered metrics at app startup time.

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 2]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
...
+configurationChange.info.CWMMC0007I.useraction=Restart the server if the libraryRef attribute was modified for the changes to take effect.
```
Comment (donbourne) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: the changes -> the change

**Review by donbourne — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 3]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
 internal.error.CWMMC0006E.explanation=The application server experienced an internal error.
 internal.error.CWMMC0006E.useraction=Gather a set of traces and open a new issue against your application server.
 
+#Information regarding libraryRef attribute taking effect only after server restart
+configurationChange.info.CWMMC0007I=CWMMC0007I: A configuration change was detected for the MicroProfile Metrics feature. If a change was made to the libraryRef attribute then a server restart is required for the change to take effect.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
configurationChange.info.CWMMC0007I=CWMMC0007I: A configuration change was detected for the MicroProfile Metrics feature. If a change was made to the libraryRef attribute, a server restart is required for the change to take effect.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 4]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
...
+configurationChange.info.CWMMC0007I.explanation=Due to the nature of how the MicroProfile Metrics feature loads the implementation binaries, a configuration change to the libraryRef attribute during runtime is not possible. Reloading the library and its contents specified by the libraryRef attribute during runtime will erase all metric data. A change to the libraryRef attribute will require a server restart for the change to take effect.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
configurationChange.info.CWMMC0007I.explanation=Due to the way that the MicroProfile Metrics feature loads the implementation binary files, a configuration change to the libraryRef attribute is not possible during run time. Reloading the library and its contents that are specified by the libraryRef attribute during run time erases all metric data. A change to the libraryRef attribute requires a server restart for the change to take effect.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 5]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
...
+configurationChange.info.CWMMC0007I.useraction=Restart the server if the libraryRef attribute was modified for the changes to take effect.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
configurationChange.info.CWMMC0007I.useraction=If the libraryRef attribute was modified, restart the server for the change to take effect.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review

**Comment by Channyboy** [comment]
#build 
#libby

**Comment by Channyboy** [comment]
#build
#libby

**Review by dmuelle — DISMISSED** [review]
LGTM thanks

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 6]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
 internal.error.CWMMC0006E.explanation=The application server experienced an internal error.
 internal.error.CWMMC0006E.useraction=Gather a set of traces and open a new issue against your application server.
 
+#Information regarding libraryRef attribute taking effect only after server restart
+configurationChange.info.CWMMC0007I=CWMMC0007I: A configuration change was detected for the MicroProfile Metrics feature. If a change was made to the libraryRef attribute, a server restart is required for the change to take effect.
+configurationChange.info.CWMMC0007I.explanation=Due to the way that the MicroProfile Metrics feature loads the implementation binary files, a configuration change to the libraryRef attribute is not possible during run time. Reloading the library and its contents that are specified by the libraryRef attribute during run time erases all metric data. A change to the libraryRef attribute requires a server restart for the change to take effect.
```
Comment (donbourne) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: The reason we can't change the library on the fly is also that the metrics are registered at app startup.

**Review by donbourne — DISMISSED** [review]
Just had one comment for you to consider.

**Comment by Channyboy** [comment]
@dmuelle Changed the explanation of the nls pops file.

**Comment by Channyboy** [comment]
#build #libby

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 7]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
 internal.error.CWMMC0006E.explanation=The application server experienced an internal error.
 internal.error.CWMMC0006E.useraction=Gather a set of traces and open a new issue against your application server.
 
+#Information regarding libraryRef attribute taking effect only after server restart
+configurationChange.info.CWMMC0007I=CWMMC0007I: A configuration change was detected for the MicroProfile Metrics feature. If a change was made to the libraryRef attribute, a server restart is required for the change to take effect.
+configurationChange.info.CWMMC0007I.explanation=A dynamic reloading of the files referenced by the libraryRef attribute would not invoke a reprocessing of MicroProfile Metrics classes as CDI beans. This would cause instability with the MicroProfile Metrics runtime.
```
Comment (donbourne) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: maybe something like...?
configurationChange.info.CWMMC0007I.explanation=MicroProfile Metrics requires a server restart in order to change libraries referenced by libraryRef.

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 8]
```diff
@@ -59,6 +59,11 @@ internal.error.CWMMC0006E=CWMMC0006E: An internal error occurred.
 internal.error.CWMMC0006E.explanation=The application server experienced an internal error.
 internal.error.CWMMC0006E.useraction=Gather a set of traces and open a new issue against your application server.
 
+#Information regarding libraryRef attribute taking effect only after server restart
+configurationChange.info.CWMMC0007I=CWMMC0007I: A configuration change was detected for the MicroProfile Metrics feature. If a change was made to the libraryRef attribute, a server restart is required for the change to take effect.
+configurationChange.info.CWMMC0007I.explanation=MicroProfile Metrics requires a server restart in order to change libraries referenced by libraryRef.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
configurationChange.info.CWMMC0007I.explanation=MicroProfile Metrics requires a server restart to change libraries referenced by libraryRef.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
minor suggestion for concision

**Comment by posmith** [comment]
L2 message review completed

**Comment by Channyboy** [comment]
#build #libby

**Review by dmuelle — DISMISSED** [review]
LGTM

**Comment by Channyboy** [comment]
#build #libby
Build after resolving merge conflict with https://github.com/OpenLiberty/open-liberty/pull/23699

**Review by zhang-ca — APPROVED** [review]
Looks good.

**Pull request merged**